### PR TITLE
feat(ONB-102): implemented a new endpoint to delete the all posts of …

### PR DESCRIPTION
**Pull request by kerimeren**
…user, crawled all users from webpage and saved them to DB.

## Description

> Description here...

### Development and testing

- [ ] Related unit test have been implemented/updated
- [ ] Application changes have been tested locally
- [ ] Related documents have been updated

### Security

- [ ] Security impact of the change has been considered
- [ ] Code follows MobileAction's security practices and policies

### Code review 

- [ ] Pull request title follows [conventional commit format](https://mobileaction.atlassian.net/wiki/spaces/ENG/pages/743112854/Conventional+Commit+Messages) and has a JIRA task id

### CODE REVIEW

**Review by tbozk — APPROVED** [review]
Eline saglik guzel kardesim :)))

File: web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java [hunk 1]
```diff
@@ -1,16 +1,42 @@
...
  * @time 17:45
  */
+
+@Transactional
 public interface IPostRepository extends JpaRepository<Post, Long>
 {
+    public final JdbcTemplate jdbcTemplate = null;
     List<Post> findAllByUserId(Long userId);
+
+    void deleteAllByUserId(Long userId);
```
Comment (edemirkirkan) [inline web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java]: You can make this query more efficient, make sql debug logs visible on your console, and try to understand the performance problem. Then modify the query

File: web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java [hunk 2]
```diff
@@ -1,16 +1,42 @@
...
 /**
  * @author sa
  * @date 17.05.2021
  * @time 17:45
  */
+
+@Transactional
 public interface IPostRepository extends JpaRepository<Post, Long>
 {
+    public final JdbcTemplate jdbcTemplate = null;
```
Comment (edemirkirkan) [inline web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java]: Why do you need such a template? What's the purpose

File: web/src/main/java/co/mobileaction/example/web/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+public class UserService implements IUserService
+{
+    private final IUserRepository userRepository;
+
+    private final IPostRepository postRepository;
+
+    private final IUserQueueService userQueueService;
+
+    @Override
+    public void saveUser(User user){
```
Comment (edemirkirkan) [inline web/src/main/java/co/mobileaction/example/web/service/UserService.java]: Make sure you format all curly braces in the project with a Allman style

File: web/src/main/resources/application-dev.properties [hunk 4]
```diff
@@ -24,7 +25,7 @@ spring.jpa.show-sql=true
 
 # Hibernate ddl auto (create, create-drop, update)
 # spring.jpa.hibernate.ddl-auto=none
-spring.jpa.hibernate.ddl-auto=create
+spring.jpa.hibernate.ddl-auto=update
```
Comment (edemirkirkan) [inline web/src/main/resources/application-dev.properties]: you can stick with the create, to be more conventional

File: worker/src/test/java/co/mobileaction/example/worker/client/CrawlerClientIntegrationTests.java [hunk 5]
```diff
@@ -17,7 +18,6 @@
  */
 @SpringBootTest
 @ExtendWith(SpringExtension.class)
-@Disabled
```
Comment (edemirkirkan) [inline worker/src/test/java/co/mobileaction/example/worker/client/CrawlerClientIntegrationTests.java]: Make sure all integration tests are disabled, readd @Disabled annotation

File: web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java (hunk 2 above)
Comment (kerimeren) [reply web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java]: It was here for some implementation try that I did not use later, I forgot to delete it.

**Pull request merged**